numpy/core/src/multiarray/numpyos.c

```diff
@@ -574,6 +574,9 @@ NumPyOS_ascii_strtold(const char *s, char** endptr)
             *endptr = (char*)s;
         }
     }
+    else if(s){
```

Comment:
Why are you checking `s`? We already dereference it on line 521, so you (and the compiler) knows it is not null.